src/Compilers/VisualBasic/Test/Semantic/Diagnostics/GetDiagnosticsTests.vb

```diff
@@ -229,8 +229,11 @@ End Namespace
             dict.Add("1", Nothing)
             Dim compilation = CreateCompilation(String.Empty, parseOptions:=New VisualBasicParseOptions().WithPreprocessorSymbols(dict))
 
+            Dim conditionalCompilationConstantNotValidFormat = VBResources.ResourceManager.GetString(
+                NameOf(ERR_ConditionalCompilationConstantNotValid),
+                EnsureEnglishUICulture.PreferredOrNull)
             CompilationUtils.AssertTheseDiagnostics(compilation, <errors>
-BC31030: Conditional compilation constant '1' is not valid: Identifier expected.
+BC31030: <%= String.Format(conditionalCompilationConstantNotValidFormat, VBResources.ERR_ExpectedIdentifier, "1") %>
```

Comment:
`CompilationUtils.AssertTheseDiagnostics` calls `CompilationUtils.ErrorText` which uses `Diagnostic.GetMessage(EnsureEnglishUICulture.PreferredOrNull)` to get the error text. `GetMessage` can translate the message but can't translate the argument. Therefore the error text ends up being a mixture of an English message and a German argument:

`BC31030: Conditional compilation constant '1' is not valid: Bezeichner erwartet.`

There are several options to solve this problem:

1. Build the same mixed english/german message as the assert expects (as I did).
2. Use the [UseCulture("en-US")] attribute to force the test to run in the en-US locale.
3. Don't use `EnsureEnglishUICulture.PreferredOrNull` in `CompilationUtils.ErrorText`

While I personally would prefer the third option it is also the one with the biggest impact. Changing `CompilationUtils.ErrorText` might cause other tests to fail (I haven't tried it, so the actual impact is unknown). To see the impact it would be useful to have `Test.cmd` passing without failures on a non-english machine as a bottom line. Therefore I would propose to merge the changes with either the first or the second solution and to follow up with another issue that removes the usages of `EnsureEnglishUICulture.PreferredOrNull` when all the PRs related to the localization problems are merged.